src/Chapter3.hs

```diff
@@ -971,6 +1084,29 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend d = fromEnum d >= 5
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:
```haskell
nextDay d = 
    if d == maxBound 
    then minBound
    else succ d
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂